### ISSUE TITLE
Serial port won't open in Windows

### ISSUE THREAD

**Issue by DRSDavidSoft** [olback/vscode-serial#1]
Does this extension support Windows serial port?

<img width="1056" alt="vscode-serial-windows" src="https://user-images.githubusercontent.com/4673812/39328744-8fb6c47e-49b1-11e8-833a-8c4d7a9992c1.png">

**Comment by olback** [comment]
It did work. I think the node module is broken since it doesn't work on Linux either anymore. Unfortunately, this extension is not my highest priority right now but I'll see what I can do about it.

**Comment by DRSDavidSoft** [comment]
@olback I appreciate your work nonetheless. Can you switch from `serialport-js` to `serialport`? I think it's better supported on each platform.

https://www.npmjs.com/package/serialport

**Comment by olback** [comment]
@DRSDavidSoft Yes, I'll look into it.

**Comment by olback** [comment]
@DRSDavidSoft Now I remember why I didn't use serialport. It does not work with Electron. Might have to write my own serialport bindings. (Might take a while...) There is another node module called [electron-serialport](https://github.com/johnny-five-io/electron-serialport) but that module seems to be depricated.

**Comment by DRSDavidSoft** [comment]
@olback Why doesn't serialport work with Electron? In the [docs](https://www.npmjs.com/package/serialport#electron) this isn't specified, it just needs compiling.

Anyways, if you plan to write your own bindings, we could fork serialport-js to work on it. Additionally, I'm working on a C++ implementation called [SerialUART](https://github.com/DRSDavidSoft/SerialUART) which could be used as the underlying bindings, as I'd like to write a new plugin for Node.js and Electron based on it to use on my Raspberry Pi, so I could make sure Windows and Linux platforms are both supported.

Thanks for the work, there's no hurry but if you'd like to put some time on vscode-serial, I'd also like in any way I can.

**Comment by olback** [comment]
I don't know how the electron stuff works with vscode and wouldn't every user have to compile it every time they update?

**Comment by DRSDavidSoft** [comment]
@olback I don't believe so, here's a similar issue which they got it to work: https://github.com/node-serialport/node-serialport/issues/1410

I think it boils down to including a pre-compiled binary (I'm not sure since I haven't worked with VS Code extensions yet, so I'll have to read the documentation on that.)

As for the [Serialport](https://github.com/node-serialport/node-serialport) npm package support in Electron, there's an example in [electron-serialport](https://github.com/johnny-five-io/electron-serialport) repo, so I guess writing new bindings _should_ not be required. In fact, [here](http://meow.noopkat.com/using-node-serialport-in-an-electron-app/)'s a simple tutorial (I haven't tested it myself yet.)

Finally, I think we could borrow some code from VS Code's official Arduino [extension](https://github.com/Microsoft/vscode-arduino) here (or at least study it since it's written completely in type script.) This specific plugin is designed to be integrated with Arduino IDE, but it should be possible to use the same UI elements for `vscode-serial`'s UI.

I'll have to study and examine each option before taking the final decision, but since this is a relatively simple project, it shouldn't be too hard. :)

**Comment by olback** [comment]
I got it to work with electron-rebuild! `./node_modules/.bin/electron-rebuild --version 1.7.12` works but my guess is that this will break with the next VS Code update? I need to fetch and pass the version automaticly to the re-build command.

**Comment by olback** [comment]
Finally got around to fixing this, v2.0.0 should be up soon.

**Comment by DRSDavidSoft** [comment]
Awesome, should it be working on both Windows and Linux platforms when v2.0.0 is released? Can't wait to test it out!

**Comment by olback** [comment]
v2.0.3 is released but does not work on Windows. Or it kind of does, it works in VS Code extension development host on windows without any problems.
